File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/DirectiveBlock.kt
package com.nouprax.markdown.core

public class DirectiveBlock internal constructor(
    public val name: String,
    /** Markup owned by the label field, never an element of [content]. */
    public val label: DirectiveLabel?,
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visitDirectiveBlock(this)
}

File: packages/es-markdown-core/src/markup/directive-block.ts
import type { MarkupBase } from "./base.js";
import type { DirectiveLabel } from "./directive-label.js";
import type { Markup } from "./markup.js";

export interface DirectiveBlock extends MarkupBase<"directiveBlock"> {
    readonly name: string | null;
    /** Markup owned by the label field, never an element of `content`. */
    readonly label: DirectiveLabel | null;
    readonly content: readonly Markup[];
}
